src/lib/rng/auto_rng/auto_rng.cpp

```diff
@@ -0,0 +1,65 @@
+/*
+* (C) 2016 Jack Lloyd
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include <botan/auto_rng.h>
+#include <botan/entropy_src.h>
+#include <botan/loadstor.h>
+#include <botan/internal/os_utils.h>
+
+#if defined(BOTAN_HAS_HMAC_DRBG)
+  #include <botan/hmac_drbg.h>
+#endif
+
+#if defined(BOTAN_HAS_HMAC_RNG)
+  #include <botan/hmac_rng.h>
+#endif
+
+#if defined(BOTAN_HAS_SYSTEM_RNG)
+  #include <botan/system_rng.h>
+#endif
+
+namespace Botan {
+
+AutoSeeded_RNG::AutoSeeded_RNG(size_t max_output_before_reseed)
+   {
+   m_rng.reset(new BOTAN_AUTO_RNG_DRBG(MessageAuthenticationCode::create(BOTAN_AUTO_RNG_HMAC),
+#if defined(BOTAN_HAS_SYSTEM_RNG)
+                                       system_rng(),
+#else
+                                       Entropy_Sources::global_sources(),
+#endif
```

Comment:
So this change means that on systems that have a system_rng (must both be supported by OS and enabled in the build), that one would be used as the seed/reseed generator for AutoSeeded_RNG and on other systems, the previous default global_sources() will be used as a fallback.

I welcome this change, it makes it easier to reason about which entropy sources are actually used and it aligns with what the current consensus seems to be about trusting the system-provided rng for day-to-day usage.